src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -61,11 +61,16 @@ namespace {
 static void GetThreadInfo(const struct k_thread * thread, void * user_data)
 {
     size_t unusedStackSize;
-    ThreadMetrics ** threadMetricsListHead = (ThreadMetrics **) user_data;
-    ThreadMetrics * threadMetrics          = (ThreadMetrics *) malloc(sizeof(ThreadMetrics));
+    ThreadMetrics ** threadMetricsListHead = static_cast<ThreadMetrics **>(user_data);
+    ThreadMetrics * threadMetrics          = Platform::New<ThreadMetrics>();
```

Comment:
Actually, `DiagnosticDataProviderImpl::ReleaseThreadMetrics` should also use `Platform::Delete`

Reply:
I was wondering how the memory was released  actually, you gave me the answer :) I'll fix that

Reply:
fixed in the last push